Realm/RLMQueryUtil.mm

```diff
@@ -55,14 +58,11 @@ void throwException(NSString *name, NSString *format, ...) {
     @throw [NSException exceptionWithName:name reason:reason userInfo:nil];
 }
 
-// check a precondition and throw an exception if it is not met
-// this should be used iff the condition being false indicates a bug in the caller
-// of the function checking its preconditions
-void RLMPrecondition(bool condition, NSString *name, NSString *format, ...) {
-    if (__builtin_expect(condition, 1)) {
-        return;
-    }
+namespace {
 
+// small helper to create the many exceptions thrown when parsing predicates
+[[gnu::cold]] [[noreturn]]
+void throwException(NSString *name, NSString *format, ...) {
```

Comment:
Did this sneak in here?

Reply:
I don't think so?
The intention was to move RLMPrecondition out of the namespace because of a conflict. When importing RLMQueryUtil.hpp, the method definition was ambiguous to method source in RLMQueryUtil.mm. RLMPrecondition is used in RLMObservation.mm.

Github's visualization of the move looks strange because` RLMPredcondition` and `[[gnu::cold]] [[noreturn]]` share four identical lines.

The most likely scenario is that moving RLMPrecondition of the namespace is not the correct way to solve ambiguity/conflict. What should be happening?

Reply:
I don't have a better answer, it simply looked out of place with the rest of the PR. Resolving.